java/src/jmri/jmrit/logixng/actions/swing/ActionListenOnBeansTableSwing.java

```diff
@@ -44,36 +44,14 @@ protected void createPanel(@CheckForNull Base object, @Nonnull JPanel buttonPane
         tableBeanPanelPanel.add(tableBeanPanel);
         panel.add(tableBeanPanelPanel);
 
+        _rowOrColumnNameComboBox = new JComboBox<>();
         _tableRowOrColumnComboBox = new JComboBox<>();
         for (TableRowOrColumn item : TableRowOrColumn.values()) {
             _tableRowOrColumnComboBox.addItem(item);
         }
         JComboBoxUtil.setupComboBoxMaxRows(_tableRowOrColumnComboBox);
         _tableRowOrColumnComboBox.addActionListener((evt) -> {
-            _rowOrColumnNameComboBox.removeAllItems();
-            NamedTable table = tableBeanPanel.getNamedBean();
-            if (table != null) {
-                if (_tableRowOrColumnComboBox.getItemAt(_tableRowOrColumnComboBox.getSelectedIndex()) == TableRowOrColumn.Column) {
-                    for (int column=0; column <= table.numColumns(); column++) {
-                        // If the header is null or empty, treat the row as a comment
-                        Object header = table.getCell(0, column);
-                        if ((header != null) && (!header.toString().isEmpty())) {
-                            _rowOrColumnNameComboBox.addItem(header.toString());
-                        }
-                    }
-                } else {
-                    for (int row=0; row <= table.numRows(); row++) {
-                        // If the header is null or empty, treat the row as a comment
-                        Object header = table.getCell(row, 0);
-                        if ((header != null) && (!header.toString().isEmpty())) {
-                            _rowOrColumnNameComboBox.addItem(header.toString());
-                        }
-                    }
-                }
-                if (action != null) {
-                    _rowOrColumnNameComboBox.setSelectedItem(action.getRowOrColumnName());
-                }
-            }
+            comboListener(action, _tableRowOrColumnComboBox, _rowOrColumnNameComboBox, tableBeanPanel);
```

Comment:
`_tableRowOrColumnComboBox` and `_rowOrColumnNameComboBox` are fields of the class so you don't need to pass these as parameters to the method.

As a general rule, I try to use underscore as the first character in the name of fields in the class. So that `_table` is a field and `table` is a local variablel.

Now when I look at the code, I noticed that the field `tableBeanPanel` should be named `_tableBeanPanel`. If you want to change it, feel free to do so.